frontend/packages/core/src/AppProvider/workflow-hydrator.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import { Grid } from "@material-ui/core";
+
+import { WorkflowStorageContext } from "../Contexts/workflow-storage-context";
+import { retrieveData } from "../Contexts/workflow-storage-context/helpers";
+import workflowStorageContextReducer from "../Contexts/workflow-storage-context/reducer";
+import type { WorkflowStorageContextProps } from "../Contexts/workflow-storage-context/types";
+import { defaultWorkflowStorageState } from "../Contexts/workflow-storage-context/types";
+import { Alert } from "../Feedback";
+import styled from "../styled";
+
+interface WorkflowHydratorProps {
+  children: React.ReactElement;
+  hydrateData: IClutch.shortlink.v1.IShareableState[];
```

Comment:
nit it might make more sense to call this something related to the type (e.g. `sharedState`)